p03-form-validations/src/Form/Form.component.js

```diff
@@ -1,20 +1,75 @@
-import React from 'react'
+import React, { useState, useEffect } from 'react'
 import { Card } from '../Shared/components/Card'
 import { Button, Input } from '../Shared/components'
-import { fieldsList } from './Form.services'
+import { fieldsList, isFieldValid } from './Form.services'
 
 import "./Form.styles.css"
 
+const initialState = fieldsList.map(field=> ({...field, ready:false, value:""}))
+
 const Form = () => {
-  
+
+  const [fieldsReady, setFieldsReady] = useState(false)
+  const [listo, setListo] = useState('Not ready to send')
+  const [fieldsForm, setFieldsForm] = useState(initialState)
+
+  useEffect(()=> {
+    setFieldsReady(true)
+    setListo('Ready to send')
+    fieldsForm.map((field) =>{
+      if (!field.ready) {
+        setFieldsReady(false)
+        setListo('Not ready to send')
+      }
+    });
+  }, [fieldsForm]);
+
+  const handleOnChange = (e) => {
+    const { name, value } = e.target
+    console.log (`campo ${name} valor ${value}`)
+
+    setFieldsForm(newFieldForms => {
+      return newFieldForms.map(field => {
+        return field.name===name ? {...field, value: value, ready : isFieldValid(value)}:field
+      })
+    })
+
+    // setFieldsForm((current) => ({
```

Comment:
Just like the console.logs, avoid the code comments in the repository. 

comments that are valid are comments to explain functionalities or descriptions
```
/**
 * function to validate the form and return a boolean value
 * params empty
 * return boolean
 */
function isValidForm () {
}

// this is a boolean to render if button is disabled or not 
const isButtonDisabled = !isValidForm()
```